SparkPost/2web2ui
# hotfix/FE-1160 - Fix pending state, fix subaccount value not clearing

Hot fix for [FE-1160](https://sparkpost.atlassian.net/browse/FE-1160)

### What Changed
- Separated pending UI state between `listPending` and `createPending` to avoid unnecessary loading state
- Fixed clear button not clearing the subaccount value

### How To Test
- Go to `/domains/create`
- Fill out the for and select 'Single Subaccount' within 'Subaccount Assignment'
- Type a subaccount, clear it with the clear button, and try to submit the form
- A validation error should occur (it wasn't occurring previously!)

### To Do
